src/services/formatting/formatting.ts

```diff
@@ -1072,7 +1087,7 @@ namespace ts.formatting {
             const nonWhitespaceColumnInFirstPart =
                 SmartIndenter.findFirstNonWhitespaceCharacterAndColumn(startLinePos, parts[0].pos, sourceFile, options);
 
-            if (indentation === nonWhitespaceColumnInFirstPart.column) {
+            if (indentation === nonWhitespaceColumnInFirstPart.column && !jsxTextStyleIndent) {

```

Comment:
Do note - if we choose to remove [this line](https://github.com/orta/TypeScript/pull/1/files#diff-270a188d87d631646eabdbdd7b019e3cR1112) because we prefer the whitespace-preserving style, then we can revert this line, too.